mongodbatlas/resource_mongodbatlas_search_index_test.go

```diff
@@ -2,53 +2,98 @@ package mongodbatlas
+				Config: testAccSearchIndexConfigBasic(projectID, indexName, databaseName, clusterNameStr, clusterTerraformStr, false),
 				Check: resource.ComposeTestCheckFunc(
 					testAccCheckSearchIndexExists(resourceName),
 					resource.TestCheckResourceAttr(resourceName, "name", indexName),
 					resource.TestCheckResourceAttr(resourceName, "project_id", projectID),
 					resource.TestCheckResourceAttr(resourceName, "cluster_name", clusterName),
-					resource.TestCheckResourceAttrSet(datasourceName, "name"),
-					resource.TestCheckResourceAttrSet(datasourceName, "project_id"),
-					resource.TestCheckResourceAttrSet(datasourceName, "collection_name"),
-					resource.TestCheckResourceAttrSet(datasourceName, "database"),
-					resource.TestCheckResourceAttrSet(datasourceName, "search_analyzer"),
-					resource.TestCheckResourceAttrSet(datasourceName, "synonyms.#"),
-					resource.TestCheckResourceAttr(datasourceName, "synonyms.#", "1"),
-					resource.TestCheckResourceAttr(datasourceName, "synonyms.0.analyzer", "lucene.simple"),
-					resource.TestCheckResourceAttr(datasourceName, "synonyms.0.name", "synonym_test"),
-					resource.TestCheckResourceAttr(datasourceName, "synonyms.0.source_collection", "collection_test"),
-					resource.TestCheckResourceAttrSet(datasourceIndexesName, "cluster_name"),
-					resource.TestCheckResourceAttrSet(datasourceIndexesName, "database"),
-					resource.TestCheckResourceAttrSet(datasourceIndexesName, "project_id"),
-					resource.TestCheckResourceAttrSet(datasourceIndexesName, "results.#"),
-					resource.TestCheckResourceAttrSet(datasourceIndexesName, "results.0.index_id"),
-					resource.TestCheckResourceAttrSet(datasourceIndexesName, "results.0.name"),
+					resource.TestCheckResourceAttr(resourceName, "database", databaseName),
+					resource.TestCheckResourceAttr(resourceName, "collection_name", collectionName),
+					resource.TestCheckResourceAttr(resourceName, "search_analyzer", searchAnalyzer),
+					resource.TestCheckResourceAttr(resourceName, "type", ""),
+
+					resource.TestCheckResourceAttr(datasourceName, "name", indexName),
+					resource.TestCheckResourceAttr(datasourceName, "project_id", projectID),
+					resource.TestCheckResourceAttr(datasourceName, "cluster_name", clusterName),
+					resource.TestCheckResourceAttr(datasourceName, "database", databaseName),
+					resource.TestCheckResourceAttr(datasourceName, "collection_name", collectionName),
+					resource.TestCheckResourceAttr(datasourceName, "mappings_dynamic", "true"),
+					resource.TestCheckResourceAttr(datasourceName, "search_analyzer", searchAnalyzer),
+					resource.TestCheckResourceAttr(datasourceName, "name", indexName),
+					resource.TestCheckResourceAttr(datasourceName, "type", ""),
+					resource.TestCheckResourceAttrSet(datasourceName, "index_id"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccSearchIndexRS_withSearchType(t *testing.T) {
```

Comment:
@marcosuma FYI this tests checks that you can explicitly pass "search" as index type.